common/src/main/java/com/microsoft/identity/common/internal/migration/TokenCacheItemMigrationAdapter.java

```diff
@@ -127,25 +128,26 @@
      *
      * @param clientId    String of the given client id.
      * @param redirectUri redirect url string of the given client id.
-     * @param cacheRecord Foci cache record.
+     * @param accountRecord account record of request
+     * @param refreshTokenRecord : refresh token record of FOCI account
      * @return true if the given client id can use the cached foci token. False, otherwise.
      * @throws ClientException
      * @throws IOException
      */
-    public static boolean tryFociTokenWithGivenClientId(@NonNull final BrokerOAuth2TokenCache brokerOAuth2TokenCache,
+    public static boolean tryFociTokenWithGivenClientId(@NonNull final OAuth2TokenCache brokerOAuth2TokenCache,
                                                         @NonNull final String clientId,
                                                         @NonNull final String redirectUri,
-                                                        @NonNull final ICacheRecord cacheRecord)
+                                                        @NonNull final RefreshTokenRecord refreshTokenRecord,
+                                                        @NonNull final IAccountRecord accountRecord)
             throws ClientException, IOException {
         final String methodName = ":tryFociTokenWithGivenClientId";
```

Comment:
Currently I believe our `logger` doesn't have functionality to get method names atleast for logs without stack trace. If we get to that functionality it would be great. To answer your question , it's been a pattern we historically followed, I am not a big fan of it personally too, but we haven't invested any cycles to change this.

Reply:
Cool.  I actually tried writing a few things to change the logging stack, but it gets to be a really big change quickly.  I'm curious why we don't use one of the more standard logging frameworks like, say, slf4j, maybe it's binary size?

Anyway, why I'm concerned a little, from https://developer.android.com/reference/android/util/Log.html :

that when you're building the string to pass into Log.d, the compiler uses a StringBuilder and at least three allocations occur: the StringBuilder itself, the buffer, and the String object. Realistically, there is also another buffer allocation and copy, and even more pressure on the gc. That means that if your log message is filtered out, you might be doing significant work and incurring significant overhead.

Reply:
IIRC, we looked at doing this awhile back and saw a lot of slowness around using `Thread.getStackTrace` to inspect the caller function. Now that threading has moved to a background thread this may be less of concern, depending on the internal mechanics of the `getStackTrace` implementation and runtime (ART vs Dalvik).